File: app/src/main/java/com/example/myapplication1/MainActivity.java
package com.example.myapplication1;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.util.Patterns;
import android.view.Gravity;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.regex.Pattern;

// Pattern pattern = Patten.compile(:*(.*)@(\\$*)$");

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setTitle("Авторизация");
    }
    @SuppressLint("SetTextI18n")
    public void onSignClick(View view) {
        Pattern pattern = Patterns.EMAIL_ADDRESS;
        Pattern pattern2 = Pattern.compile("^(?=.*[A-Z])(?=.*[!@#$%^&*?])(?=.{7,})\\S+$");

        EditText editLogin = findViewById(R.id.editLogin);
        EditText editPassword = findViewById(R.id.editPassword);
        TextView textIn = findViewById(R.id.textView2);

        String editLog = editLogin.getText().toString();
        String editPas = editPassword.getText().toString();

        if (editLog.isEmpty() || editPas.isEmpty()) {
            textIn.setGravity(Gravity.CENTER);
            textIn.setText("Entering an email and password is required!");
        }else {
            if (!pattern.matcher(editLog).matches() || !pattern2.matcher(editPas).matches()) {
                textIn.setGravity(Gravity.CENTER);
                textIn.setText("Incorrect email address or password");
            }else {
                textIn.setGravity(Gravity.CENTER);
                textIn.setText("Successfully!");
                Intent i = new Intent(this, SecondActivity.class);
                i.putExtra("log", editLog);
                startActivity(i);
            }
        }
    }
}

File: app/src/main/java/com/example/firebaseproject/MainActivity.kt
package com.example.firebaseproject

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Patterns
import android.widget.Toast
import com.example.firebaseproject.databinding.ActivityMainBinding
import com.google.firebase.auth.AuthResult
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.firestore.CollectionReference
import com.google.firebase.firestore.FirebaseFirestore

class MainActivity : AppCompatActivity() {
    lateinit var binding: ActivityMainBinding
    var firebaseAuth = FirebaseAuth.getInstance()
    var db: FirebaseFirestore = FirebaseFirestore.getInstance()

    var userReference: CollectionReference = db.collection("users")

    override fun onCreate(savedInstanceState:Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        if (firebaseAuth.currentUser?.email != null) {
            startActivity(Intent(this, UserMainActivity::class.java))
        }

        val emailInput = binding.emailInput
        val passwordInput = binding.passwordInput

        binding.logButton.setOnClickListener {
            startActivity(Intent(this, LoginActivity::class.java))
        }

        binding.regButton.setOnClickListener {
            val email: String = emailInput.text.toString().trim()
            val password: String = passwordInput.text.toString().trim()

            if (!Patterns.EMAIL_ADDRESS.matcher(email).matches())
                emailInput.error = "Неверный формат почты"
            else if (password.length < 6)
                passwordInput.error = "Пароль слишком короткий"
            else registerUser(email, password)
        }
    }

    private fun registerUser(email: String, password: String) {
        firebaseAuth.createUserWithEmailAndPassword(email, password)
            .addOnSuccessListener { authResult: AuthResult ->
                val fireBaseUser: FirebaseUser? = authResult.user

                val newUser = User(fireBaseUser!!.uid, fireBaseUser.email)
                userReference.add(newUser)
                    .addOnSuccessListener {
                        Toast.makeText(this, "Пользователь " + fireBaseUser.email + " зарегистрирован",
                            Toast.LENGTH_SHORT).show()
                    }
            }
            .addOnFailureListener {
                Toast.makeText(this, it.message, Toast.LENGTH_SHORT).show()
            }
    }
}